src/plugins/intel_cpu/src/nodes/executors/fullyconnected_implementations.cpp

```diff
@@ -0,0 +1,315 @@
+            Dim widthInConv = inDims[inRank - 2];
+            Dim K = inDims[inRank - 1];
+            Dim N = weightDims[0];
+
+            const auto& weightsSize = weiMemSize(key);
+            // Disable Conv1x1 when weight size >= 16M to avoid different weight layout when having different input
+            // activation shapes. As a consuquence, peak memory consumption in LLM can be decreased.
+            VERIFY(weightsSize < (16 * 1 << 20), " weights size is to big");
+            VERIFY(widthInConv >= 2 && widthInConv <= 3136 && K >= 96 && K <= 4096 && N >= 96 && N <= K * 4,
+                   HEURISTICS_MISMATCH);
+
+            return true;
+        },
+        // create
+        [](const FCConfig& key, const MemoryArgs& memory, ExecutorContext::CPtr context) {
+            struct ConvolutionInstantiator {
+                std::shared_ptr<DnnlConvolutionPrimitive> operator()(
+                    const MemoryDescArgs& descs,
+                    const FCAttrs& attrs,
+                    const ExecutorContext::CPtr context,
+                    std::shared_ptr<DnnlShapeAgnosticData> shareAgnosticData) const {
+                    ConvAttrs convAttrs{attrs.withBias};
+                    return DefaultInstantiator<DnnlConvolutionPrimitive, ConvAttrs, DnnlShapeAgnosticData>{}(
+                        descs,
+                        convAttrs,
+                        context,
+                        shareAgnosticData);
+                }
+            };
+
+            return std::make_shared<
+                DnnlFCExecutor<DnnlConvolutionPrimitive, FCAttrs, DnnlShapeAgnosticData, ConvolutionInstantiator>>(
+                key,
+                memory,
+                context,
+                false);
+        })
+    OV_CPU_INSTANCE_DNNL(
+        // @todo executor type should not be a part of the name (can be appended if necessary)
+        "fullyconnected_dnnl",  // name
+        ExecutorType::Dnnl,     // type
+        OperationType::FullyConnected,
+        ShapeTolerance::Dependant,
+        // isSupported
+        [](const FCConfig& key) -> bool {
+            return true;
+        },
+        // isFullyCompliant
+        [](const FCConfig& key) {
+            return isFullyCompliantCommon(key, dnnlFCTypeMapping, dnnlFCLayoutConfig);
+        },
+        // isShapeSuitable
+        [](const FCConfig& key) -> bool {
+            return true;
+        },
+        // create
+        [](const FCConfig& key, const MemoryArgs& memory, ExecutorContext::CPtr context) {
+            return std::make_shared<DnnlFCExecutor<DnnlFCPrimitive, FCAttrs, DnnlShapeAgnosticData>>(key,
+                                                                                                     memory,
+                                                                                                     context,
+                                                                                                     false);
```

Comment:
@todo This must be set to true for ARM to ensure weights prepacking in scope of compile_model